include/RAJA/pattern/launch/launch_core.hpp

```diff
@@ -214,24 +214,57 @@ class LaunchContext
 template <typename LAUNCH_POLICY>
 struct LaunchExecute;
 
-//Policy based launch without name argument
-template <typename LAUNCH_POLICY, typename BODY>
-void launch(LaunchParams const &params, BODY const &body)
+//Policy based launch with support to new reducers...
+template <typename LAUNCH_POLICY, typename ... ReduceParams>
+void launch(LaunchParams const &launch_params, const char *kernel_name, ReduceParams&&... rest_of_launch_args)
 {
-  launch<LAUNCH_POLICY>(params, nullptr, body);
+
+  //Get reducers
+  auto reducers = expt::make_forall_param_pack(std::forward<ReduceParams>(rest_of_launch_args)...);
+
+  auto&& launch_body = expt::get_lambda(std::forward<ReduceParams>(rest_of_launch_args)...);
+
+  //Take the first policy as we assume the second policy is not user defined.
+  //We rely on the user to pair launch and loop policies correctly.
+  util::PluginContext context{util::make_context<typename LAUNCH_POLICY::host_policy_t>()};
+  util::callPreCapturePlugins(context);
+
+  using RAJA::util::trigger_updates_before;
+  auto p_body = trigger_updates_before(launch_body);
+
+  util::callPostCapturePlugins(context);
+
+  util::callPreLaunchPlugins(context);
+
+  using launch_t = LaunchExecute<typename LAUNCH_POLICY::host_policy_t>;
+
+  using Res = typename resources::get_resource<typename LAUNCH_POLICY::host_policy_t>::type;
+
+  launch_t::exec(Res::get_default(), launch_params, kernel_name, p_body, reducers);
+
+  util::callPostLaunchPlugins(context);
 }
 
-//Policy based launch
-template <typename LAUNCH_POLICY, typename BODY>
-void launch(LaunchParams const &params, const char *kernel_name, BODY const &body)
+
+//Duplicate of code above on account that we need to support the case in which a kernel_name is not given
+template <typename LAUNCH_POLICY, typename ... ReduceParams>
+void launch(LaunchParams const &launch_params, ReduceParams&&... rest_of_launch_args)
```

Comment:
I worry that this overload may be a better match in some situations for calls taking kernel_name than the overload above. For example passing in a non-const char*.

Reply:
Ah, OK. I was unaware of that. It should be simple to add and a good thing for consistency.

Reply:
I agree. I would prefer if we use the KernelName object introduced in the new forall interface; however its currently broken and is missing support for non-gpu backends (https://github.com/LLNL/RAJA/issues/1566) . I would like to address it in a follow up PR unifying the launch and forall backends.

Reply:
@artv3 if there isn't one already, please add an issue for that and tag it with  the milestone for the next release. Thank you.